Currently fails to build, when I give no profile argument to `nixos-rebuild switch`:

```
created 10635 symlinks in user environment                                                                                                                                  
install-info: warning: no info dir entry in `/nix/store/1zjrj9rna9gs16d36kc53l6dzy9kkxab-system-path/share/info/time.info'                                                  
building path(s) ‘/nix/store/w9kjfxkm8hfvrfqlxxr0hilfivxl81cn-dbus-1’                                                                                                       
building path(s) ‘/nix/store/y8s1j7cv8maf7i7crjajdmvci4lg5x2v-unit-polkit.service’                                                                                          
building path(s) ‘/nix/store/jqp8b44i1lx82i4h4fpialqxl1qdssdd-unit-systemd-fsck-.service’                                                                                   
building path(s) ‘/nix/store/0p423ar4pj10jxa5277pprm0bn5paavr-unit-dbus.service’                                                                                            
building path(s) ‘/nix/store/6wx3l0v74p32f8icrq9vmkw370m7kdqw-system-units’                                                                                                 
building path(s) ‘/nix/store/h4z06sv1nys9sl6mrndk749np9d76bh2-user-units’                                                                                                   
building path(s) ‘/nix/store/gyxmp1np5zlk7vnilz71qkc3hr5rppmm-etc’                                                                                                          
building path(s) ‘/nix/store/m0w6z41fsj1qdpmw34bxcpq729qngpl8-nixos-system-turingmachine-17.09.git.5334e85’                                                                 
Traceback (most recent call last):                                                                                                                                          
  File "/nix/store/xns7s94x8a7z24hi9qd9jzrsl445sscz-systemd-boot-builder.py", line 182, in <module>                                                                         
    main()                                                                                                                                                                  
  File "/nix/store/xns7s94x8a7z24hi9qd9jzrsl445sscz-systemd-boot-builder.py", line 165, in main                                                                             
    for profile in get_profiles():                                                                                                                                          
  File "/nix/store/xns7s94x8a7z24hi9qd9jzrsl445sscz-systemd-boot-builder.py", line 127, in get_profiles                                                                     
    for x in os.listdir("/nix/var/nix/profiles/system-profiles/")                                                                                                           
FileNotFoundError: [Errno 2] No such file or directory: '/nix/var/nix/profiles/system-profiles/'                                                                            
warning: error(s) occurred while switching to the new configuration 
```

--- a/nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py
+++ b/nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py
@@ -107,7 +107,7 @@ def remove_old_entries(gens):
     for gen in gens:
         known_paths.append(copy_from_profile(*gen, "kernel", True))
         known_paths.append(copy_from_profile(*gen, "initrd", True))
-    for path in glob.iglob("@efiSysMountPoint@/loader/entries/nixos-*-generation-[1-9]*.conf"):
+    for path in glob.iglob("@efiSysMountPoint@/loader/entries/nixos*-generation-[1-9]*.conf"):
         try:
             if rex_profile.match(path):
                 prof = rex_profile.sub(r"\1", path)
@@ -123,9 +123,12 @@ def remove_old_entries(gens):
             os.unlink(path)
 
 def get_profiles():
-    return [x
+    if os.path.isdir("/nix/var/nix/profiles/system-profiles/"):
+        return [x
             for x in os.listdir("/nix/var/nix/profiles/system-profiles/")
             if not x.endswith("-link")]
+    else:
+        return []
 
 def main():
     parser = argparse.ArgumentParser(description='Update NixOS-related systemd-boot files')

